homeassistant/components/template/weather.py

```diff
@@ -0,0 +1,220 @@
+"""Template platform that aggregates meteorological data."""
+import voluptuous as vol
+
+from homeassistant.components.weather import (
+    ATTR_CONDITION_CLOUDY,
+    ATTR_CONDITION_EXCEPTIONAL,
+    ATTR_CONDITION_FOG,
+    ATTR_CONDITION_HAIL,
+    ATTR_CONDITION_LIGHTNING,
+    ATTR_CONDITION_LIGHTNING_RAINY,
+    ATTR_CONDITION_PARTLYCLOUDY,
+    ATTR_CONDITION_POURING,
+    ATTR_CONDITION_RAINY,
+    ATTR_CONDITION_SNOWY,
+    ATTR_CONDITION_SNOWY_RAINY,
+    ATTR_CONDITION_SUNNY,
+    ATTR_CONDITION_WINDY,
+    ATTR_CONDITION_WINDY_VARIANT,
+    ENTITY_ID_FORMAT,
+    WeatherEntity,
+)
+from homeassistant.const import CONF_NAME, CONF_UNIQUE_ID
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.helpers.reload import async_setup_reload_service
+
+from .const import DOMAIN, PLATFORMS
+from .template_entity import TemplateEntity
+
+CONDITION_CLASSES = {
+    ATTR_CONDITION_CLOUDY,
+    ATTR_CONDITION_FOG,
+    ATTR_CONDITION_HAIL,
+    ATTR_CONDITION_LIGHTNING,
+    ATTR_CONDITION_LIGHTNING_RAINY,
+    ATTR_CONDITION_PARTLYCLOUDY,
+    ATTR_CONDITION_POURING,
+    ATTR_CONDITION_RAINY,
+    ATTR_CONDITION_SNOWY,
+    ATTR_CONDITION_SNOWY_RAINY,
+    ATTR_CONDITION_SUNNY,
+    ATTR_CONDITION_WINDY,
+    ATTR_CONDITION_WINDY_VARIANT,
+    ATTR_CONDITION_EXCEPTIONAL,
+}
+
+CONF_WEATHER = "weather"
+CONF_TEMPERATURE_TEMPLATE = "temperature_template"
+CONF_HUMIDITY_TEMPLATE = "humidity_template"
+CONF_CONDITION_TEMPLATE = "condition_template"
+CONF_PRESSURE_TEMPLATE = "pressure_template"
+CONF_WIND_SPEED_TEMPLATE = "wind_speed_template"
+CONF_FORECAST_TEMPLATE = "forecast_template"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_CONDITION_TEMPLATE): cv.template,
+        vol.Required(CONF_TEMPERATURE_TEMPLATE): cv.template,
+        vol.Optional(CONF_HUMIDITY_TEMPLATE): cv.template,
+        vol.Optional(CONF_PRESSURE_TEMPLATE): cv.template,
+        vol.Optional(CONF_WIND_SPEED_TEMPLATE): cv.template,
+        vol.Optional(CONF_FORECAST_TEMPLATE): cv.template,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Template weather."""
+    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+
+    name = config[CONF_NAME]
+    condition_template = config[CONF_CONDITION_TEMPLATE]
+    temperature_template = config[CONF_TEMPERATURE_TEMPLATE]
+    humidity_template = config.get(CONF_HUMIDITY_TEMPLATE)
```

Comment:
Currently humidity is required.  We can swap it back if that changes.
```suggestion
    humidity_template = config[CONF_HUMIDITY_TEMPLATE]
```